File: Selenium/Activities/Activity4.py
from selenium import webdriver
from selenium.webdriver.firefox.service import Service as FirefoxService
from webdriver_manager.firefox import GeckoDriverManager
from selenium.webdriver.common.by import By

# Set up the Firefox Driver with WebDriverManger
service = FirefoxService(GeckoDriverManager().install())

# Start the Driver
with webdriver.Firefox(service=service) as driver:

    # Open the link in browser
    driver.get("https://training-support.net/selenium/target-practice")

    # Print page title
    print("Page Title is: " + driver.title)

    # Get 3rd header text
    h3 = driver.find_element(By.XPATH, "//h3[@id='third-header']")
    print("3rd header is: " + h3.text)

    # Get 5th header color
    h5 = driver.find_element(By.XPATH, "//h5[@class='ui green header']")
    print("5th header color: " + h5.value_of_css_property("color"))

    # Violet button classes
    violet = driver.find_element(By.XPATH, "//button[text()='Violet']")
    print("Violet button classes: " + violet.get_attribute("class"))

    # Grey button text
    grey = driver.find_element(By.XPATH, "//button[text()='Grey']")
    print("Grey button text: " + grey.text)

    driver.quit()

File: Selenium/Activities/Activity4.java
package activities;

import io.github.bonigarcia.wdm.WebDriverManager;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;

public class Activity4 {
    public static void main(String[] args) {
        // Installing Firefox driver
        WebDriverManager.firefoxdriver().setup();

        // Suppressing browser log
        System.setProperty(FirefoxDriver.SystemProperty.BROWSER_LOGFILE, "NULL");
        // Initialize firefox driver
        WebDriver driver = new FirefoxDriver();

        // Open a page
        driver.get("https://training-support.net/selenium/target-practice");

        // Print page title
        System.out.println("Page Title is: " + driver.getTitle());

        // Get 3rd header text
        WebElement h3 = driver.findElement(By.xpath("//h3[@id='third-header']"));
        System.out.println("3rd header is: " + h3.getText());

        // Get 5th header color
        WebElement h5 = driver.findElement(By.xpath("//h5[@class='ui green header']"));
        System.out.println("5th header color: " + h5.getCssValue("color"));

        // Violet button classes
        WebElement violet = driver.findElement(By.xpath("//button[text()='Violet']"));
        System.out.println("Violet button classes: " + violet.getAttribute("class"));

        //Grey button text
        WebElement grey = driver.findElement(By.xpath("//button[text()='Grey']"));
        System.out.println("Grey button text: " + grey.getText());

        driver.quit();
    }
}
